GitLog.py

```diff
@@ -0,0 +1,269 @@
+""" 
+
+`git log` extension
+
+It will detect every GIT in the directories
+`~/*` and `~/git/*`. It's conifigurable via
+`~/.config/albert/org.albert.extension.gitlog/config.yaml`.
+
+Use this as template for `config.yaml`:
+
+---
+git-directories:
+ - ~/git/*/*
+ - ~/src/*/*
+
+"""
+
+import os
+import subprocess
+from shutil import which
+from albertv0 import *
+import re
+
+import yaml
+import os.path
+import glob
+
+__iid__ = "PythonInterface/v0.1"
+__prettyname__ = "Git-Log-Interface"
+__version__ = "1.0"
+__trigger__ = "git "
+__author__ = "Erik Wasser"
+__dependencies__ = []
+
+#   This is our standard configuration, if we don't file a file
+#   `~/.config/albert/org.albert.extension.gitlog/config.yaml`.
+
+CONFIG_FILENAME = os.path.expanduser('~/.config/albert/org.albert.extension.gitlog/config.yaml')
+
+config = {
+    'git-directories': [
+        '~/*',
+        '~/git/*',
+    ],
+}
+
+########################################################################
+
+def discover_git_directories(directory):
+
+    prefix = os.path.expanduser('~') + '/'
+
+    for i in glob.glob(directory):
+       source = i
+       target = i
+       test_path = os.path.join(i, '.git')
+       if os.path.isdir(test_path):
+           if target[:len(prefix)] == prefix:
+               source = source[len(prefix):]
+
+           yield source, target
+
+import configparser
+import os.path
+
+class GitConfig:
+    def __init__(self, **kwargs):
+        super().__init__()
+        self.config = configparser.ConfigParser()
+
+    def read(self, directory):
+        configuration_file = os.path.join(directory, '.git/config')
+        self.config.read(configuration_file)
+
+### SHAMELESS CODE REUSE FROM https://github.com/johnkchiu/GitLogParser/blob/master/gitLogParser.py
+
+import sys
+import re
+import subprocess
+
+class GitLogParser:
+    def __init__(self, **kwargs):
+        super().__init__()
+        # array to store dict of commit data
+        self.commits = []
+
+    def parse(self, commitLines):
+        # dict to store commit data
+        commit = {}
+        # iterate lines and save
+        for nextLine in commitLines:
+            if nextLine == '' or nextLine == '\n':
+                # ignore empty lines
+                pass
+            elif bool(re.match('commit', nextLine, re.IGNORECASE)):
+                # commit xxxx
+                if len(commit) != 0:		## new commit, so re-initialize
+                    self.commits.append(commit)
+                    commit = {}
+                commit = {'hash' : re.match('commit (.*)', nextLine, re.IGNORECASE).group(1) }
+            elif bool(re.match('merge:', nextLine, re.IGNORECASE)):
+                # Merge: xxxx xxxx
+                pass
+            elif bool(re.match('author:', nextLine, re.IGNORECASE)):
+                # Author: xxxx <xxxx@xxxx.com>
+                m = re.compile('Author: (.*) <(.*)>').match(nextLine)
```

Comment:
Why do you compile the regexp here and don't compile it the lines before or after?
Skip the compiling. According to https://stackoverflow.com/a/452143 it Regexes are cached internally.